core/src/test/java/net/opentsdb/storage/TestMockDataStore.java

```diff
@@ -0,0 +1,212 @@
+import net.opentsdb.storage.MockDataStore.MockSpan;
+import net.opentsdb.utils.Config;
+
+public class TestMockDataStore {
+
+  private TSDB tsdb;
+  private Config config;
+  private QueryContext context;
+  
+  @Before
+  public void before() throws Exception {
+    tsdb = mock(TSDB.class);
+    config = new Config(false);
+    context = mock(QueryContext.class);
+    when(tsdb.getConfig()).thenReturn(config);
+    
+    config.overrideConfig("MockDataStore.timestamp", "1483228800000");
+  }
+  
+  @Test
+  public void initialize() throws Exception {
+    MockDataStore mds = new MockDataStore();
+    mds.initialize(tsdb).join();
+    assertEquals(4 * 4 * 4, mds.getDatabase().size());
+    
+    for (Map<TypeToken<?>, MockSpan<?>> types : mds.getDatabase().values()) {
+      MockSpan<?> spans = types.get(NumericType.TYPE);
+      assertEquals(24, spans.rows().size());
+      
+      long ts = 1483228800000L;
+      for (MockRow<?> row : spans.rows()) {
+        assertEquals(ts, row.base_timestamp);
+        
+        TimeSeriesIterator<NumericType> iterator = 
+            (TimeSeriesIterator<NumericType>) row.iterator();
+        int count = 0;
+        while (iterator.status() == IteratorStatus.HAS_DATA) {
+          assertEquals(ts + (count * 60000), iterator.next().timestamp().msEpoch());
+          count++;
+        }
+        ts += MockDataStore.ROW_WIDTH;
+        assertEquals(60, count);
+      }
+      assertEquals(1483315200000L, ts);
+    }
+  }
+  
+  @Test
+  public void query() throws Exception {
+    MockDataStore mds = new MockDataStore();
+    mds.initialize(tsdb).join();
+    
+    TimeSeriesQuery query = TimeSeriesQuery.newBuilder()
+        .setTime(Timespan.newBuilder()
+            .setStart("1483228800000")
+            .setEnd("1483236000000"))
+        .addMetric(Metric.newBuilder()
+            .setMetric("sys.cpu.user")
+            .setFilter("f1")
+            .setId("m1"))
+        .addMetric(Metric.newBuilder()
+            .setMetric("web.requests")
+            .setFilter("f1")
+            .setId("m2"))
+        .addFilter(Filter.newBuilder()
+            .setId("f1")
+            .addFilter(TagVFilter.newBuilder()
+                .setFilter("web01")
+                .setType("literal_or")
+                .setTagk("host")
+                )
+            )
+        .build();
+    
+    QueryExecution<IteratorGroups> ex = mds.runTimeSeriesQuery(context, query, null);
+    IteratorGroups results = ex.deferred().join();
+    assertEquals(8, results.flattenedIterators().size());
+    
+    IteratorGroup group = results.group(new SimpleStringGroupId("m1"));
+    List<TimeSeriesIterator<?>> iterators = group.flattenedIterators();
+    assertEquals(4, iterators.size());
+    for (final TimeSeriesIterator<?> it : iterators) {
+      TimeSeriesIterator<NumericType> iterator = (TimeSeriesIterator<NumericType>) it;
+      long ts = 1483228800000L;
+      while (iterator.status() == IteratorStatus.HAS_DATA) {
```

Comment:
I wish we can replace IteratorStatus with a Java iterator

Reply:
I will shortly :)